libs/coin-modules/coin-tezos/src/logic/validateIntent.ts

```diff
@@ -0,0 +1,140 @@
+import { TransactionIntent, TransactionValidation } from "@ledgerhq/coin-framework/api/types";
+import {
+  InvalidAddress,
+  RecipientRequired,
+  RecommendUndelegation,
+  NotEnoughBalance,
+  NotEnoughBalanceToDelegate,
+  AmountRequired,
+  InvalidAddressBecauseDestinationIsAlsoSource,
+} from "@ledgerhq/errors";
+import { validateAddress, ValidationResult } from "@taquito/utils";
+import api from "../network/tzkt";
+import { estimateFees } from "./estimateFees";
+import { InvalidAddressBecauseAlreadyDelegated } from "../types/errors";
+import { mapIntentTypeToTezosMode } from "../utils";
+
+export async function validateIntent(intent: TransactionIntent): Promise<TransactionValidation> {
+  // central place to validate amounts/fees for generic bridge
+  const errors: Record<string, Error> = {};
+  const warnings: Record<string, Error> = {};
+  let estimatedFees: bigint;
+  let amount: bigint;
+  let totalSpent: bigint;
+
+  // basic recipient validation for send
+  if (intent.type === "send") {
+    if (!intent.recipient) {
+      errors.recipient = new RecipientRequired("");
+    } else if (validateAddress(intent.recipient) !== ValidationResult.VALID) {
+      errors.recipient = new InvalidAddress(undefined, { currencyName: "Tezos" });
+    } else if (intent.sender === intent.recipient) {
+      errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+    }
+  }
+
+  // for generic expectations: empty amount should trigger AmountRequired
+  if (intent.type === "send" && intent.amount === 0n && !intent.useAllAmount) {
+    errors.amount = new AmountRequired();
+  }
+
+  // reject negative amounts explicitly
+  if (intent.type === "send" && intent.amount < 0n) {
+    errors.amount = new NotEnoughBalance();
+  }
+
+  if (
+    errors.recipient ||
+    errors.amount instanceof AmountRequired ||
```

Comment:
this can be enough ?
```suggestion
    errors.amount
```

Reply:
Fixed